bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v1/ExperienceFragmentImpl.java

```diff
@@ -0,0 +1,183 @@
+    adapters = {ExperienceFragment.class, ComponentExporter.class },
+    resourceType = {ExperienceFragmentImpl.RESOURCE_TYPE_V1 })
+@Exporter(name = ExporterConstants.SLING_MODEL_EXPORTER_NAME,
+    extensions = ExporterConstants.SLING_MODEL_EXTENSION)
+public class ExperienceFragmentImpl implements ExperienceFragment {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ExperienceFragmentImpl.class);
+
+    public static final String RESOURCE_TYPE_V1 = "core/wcm/components/experiencefragment/v1/experiencefragment";
+
+    public static final String PATH_DELIMITER = "/";
+    public static final char PATH_DELIMITER_CHAR = '/';
+
+    private static final String CONTENT_ROOT = "/content";
+    private static final String EXPERIENCE_FRAGMENTS_ROOT = "/content/experience-fragments";
+
+    @Self
+    private SlingHttpServletRequest request;
+
+    @ScriptVariable
+    private Page currentPage;
+
+    @ScriptVariable
+    private ValueMap properties;
+
+    @ScriptVariable
+    private Style currentStyle;
+
+    @ValueMapValue(name = ExperienceFragment.PN_FRAGMENT_PATH, injectionStrategy = InjectionStrategy.OPTIONAL)
+    private String fragmentPath;
+
+    private int localizationDepth;
+    private String localizationRoot;
+    private ResourceResolver resolver;
+    private PageManager pageManager;
+
+    @PostConstruct
+    private void initModel() {
+        localizationRoot = properties.get(ExperienceFragment.PN_LOCALIZATION_ROOT, currentStyle.get(ExperienceFragment.PN_LOCALIZATION_ROOT, String.class));
+        localizationDepth = properties.get(ExperienceFragment.PN_LOCALIZATION_DEPTH, currentStyle.get(ExperienceFragment.PN_LOCALIZATION_DEPTH, 1));
+        resolver = request.getResourceResolver();
+        pageManager = currentPage.getPageManager();
+    }
+
+    @Override
+    public String getFragmentPath() {
+        return fragmentPath;
+    }
+
+    @Override
+    public String getLocalizedFragmentPath() {
+        String localizedFragmentPath = null;
+        Page xfPage = pageManager.getPage(fragmentPath);
+        if (xfPage != null) {
+            Page xfLocalizedPage = getXfLocalizedPage(xfPage, currentPage);
+            if (xfLocalizedPage != null) {
+                localizedFragmentPath = xfLocalizedPage.getContentResource().getPath();
+            } else {
+                localizedFragmentPath = xfPage.getContentResource().getPath();
+            }
+        } else {
+            LOGGER.debug("Experience Fragment variation not found at path:{}", fragmentPath);
+        }
+        return localizedFragmentPath;
+    }
+
+    @NotNull
+    @Override public String getExportedType() {
+        return request.getResource().getResourceType();
+    }
+
+    /**
+     * Returns the localized page of the experience fragment page, based on the content page and the localization properties.
+     *
+     * @param xfPage the experience fragment page
+     * @param sitePage the content page
+     * @return
```

Comment:
Use ```javax.annotation.CheckForNull```
I advice to document the return value and the meaning of a ```NULL```
Though this is private, it helps anybody unfamiliar to the class to understand the meaning without reading all code